compiler/env/OMRAOTMethodHeader.cpp

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "env/AOTMethodHeader.hpp"
+#include <stdlib.h>
+#include <string.h>
+
+TR::AOTMethodHeader* OMR::AOTMethodHeader::self(){
+    return reinterpret_cast<TR::AOTMethodHeader*> (this);
+}
+
+void* OMR::AOTMethodHeader::serialize(){
+    uintptrj_t allocSize = this->sizeOfSerializedVersion();
+    uint8_t* buffer = (uint8_t*) malloc(allocSize);
```

Comment:
How is this memory managed?  Who frees the buffer?  Possible memory leak?

Reply:
Right, this should be freed after we send it to the shared cache. Also, I feel it should be refactored to use TR compiler memory management infrastructure. I will address that in later commits.

Reply:
This was addressed by introducing a memory allocation call to AOTStorageInterface